src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -315,6 +315,19 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                     }
                 }
 
+                Type? iCollectionInterfaceType = GetICollectionInterfaceType(type);
+                if (iCollectionInterfaceType is not null)
```

Comment:
This feels like it is duplicated with

https://github.com/dotnet/runtime/blob/9df6ea21007319967975dc9985413bb6518287da/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L371-L374

I think these should be combined.

I wonder if we can just delete this code here, and

1. Let the linked code handle the case when an object already exists.
2. Add back in the "make a new collection instance" code inside the `if (bindingPoint.Value is null)` check. See 
https://github.com/dotnet/runtime/blob/15d285f325f5ad209cd2ede3262e5f65ccf1bd73/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L367-L374